tests/sharingTest.js

```diff
@@ -36,190 +44,494 @@ describe('Main: Currently testing file/folder sharing,', function () {
     '文件.txt': 19
   }
   const testFolderShareID = 9
+  const testFolderShareToken = '6Bvt5fDajGxiV3P'
 
   // CONSTANTS FROM lib/public/constants.php
   const OCS_PERMISSION_READ = 1
   const OCS_PERMISSION_UPDATE = 2
-  const OCS_PERMISSION_CREATE = 4
   const OCS_PERMISSION_SHARE = 16
 
-  const getSharesInteraction = function (provider, requestName, shareType, file) {
-    const iteration = testFiles.indexOf(file)
-    const fileId = config.testFilesId[iteration]
-    const fileToken = config.testFilesToken[iteration]
-    return provider.uponReceiving(`as '${username}', a GET request to ${requestName}`)
+  const givenUserExists = (provider, username, password) => {
+    return provider
+      .given('the user is recreated', { username, password })
+  }
+  const givenGroupExists = (provider, group) => {
+    return provider
+      .given('group exists', { groupName: group })
+  }
+  const givenFileExists = (provider, username, password, resource) => {
+    return provider
+      .given('file exists', { username, password, fileName: encodeURIComponent(resource) })
+  }
+  const givenFolderExists = (provider, username, password, resource) => {
+    return provider
+      .given('folder exists', { username, password, folderName: resource })
+  }
+  const givenResourceShared = async (provider, username, password, shareType, resource, shareWith) => {
+    return provider
+      .given('resource is shared', { username, password, resource, shareType, shareWith })
+  }
+  let shareId, shareToken
+  const createFileOrFolder = (provider, username, password, resource, resourceType) => {
```

Comment:
lets be consistent with the naming
```suggestion
  const givenfFileFolderExists = (provider, username, password, resource, resourceType) => {
```